src/codeyCommand.ts

```diff
@@ -10,72 +33,256 @@ export type SapphireMessageExecuteType = (
+  /** Options for the Codey command */
+  options: CodeyCommandOption[] = [];
+  /** Subcommands under the CodeyCommand */
+  subcommandDetails: { [name: string]: CodeyCommandDetails } = {};
+}
+
+/** Sets the command subcommand in the slash command builder */
+const setCommandSubcommand = (
+  builder: SlashCommandBuilder,
+  subcommandDetails: CodeyCommandDetails
+): SlashCommandSubcommandsOnlyBuilder => {
+  const subcommandBuilder = new SlashCommandSubcommandBuilder();
+  subcommandBuilder.setName(subcommandDetails.name);
+  subcommandBuilder.setDescription(subcommandDetails.description);
+  for (const commandOption of subcommandDetails.options) {
+    setCommandOption(subcommandBuilder, commandOption);
+  }
+  return builder.addSubcommand(subcommandBuilder);
+};
+
+/** The codey command class */
+export class CodeyCommand extends SapphireCommand {
+  /** The details of the command */
+  details!: CodeyCommandDetails;
+
+  // Get slash command builder
+  public configureSlashCommandBuilder(builder: SlashCommandBuilder): SlashCommandBuilder {
+    builder.setName(this.details.name);
+    builder.setDescription(this.details.description);
+    for (const commandOption of this.details.options) {
+      setCommandOption(builder, commandOption);
+    }
+    for (const subcommandName in this.details.subcommandDetails) {
+      setCommandSubcommand(builder, this.details.subcommandDetails[subcommandName]);
+    }
+    return builder;
+  }
 
   // Register application commands
   public override registerApplicationCommands(registry: ChatInputCommand.Registry): void {
-    registry.registerChatInputCommand((builder) => builder.setName(this.name).setDescription(this.description));
+    // This ensures any new changes are made to the slash commands are made
+    ApplicationCommandRegistries.setDefaultBehaviorWhenNotIdentical(RegisterBehavior.Overwrite);
+    // We need to do this because TS is weird
+    registry.registerChatInputCommand((builder) =>
+      this.configureSlashCommandBuilder(<SlashCommandBuilder>(<unknown>builder))
+    );
   }
 
   // Regular command
-  public async messageRun(message: Message): Promise<Message<boolean> | undefined> {
+  public async messageRun(message: Message, commandArgs: Args): Promise<Message<boolean> | undefined> {
     const { client } = container;
+
+    // Get command arguments
+    const args: CodeyCommandArguments = {};
+    for (const commandOption of this.details.options!) {
+      args[commandOption.name] = <CodeyCommandArgumentValueType>(
+        await commandArgs.pick(<keyof ArgType>commandOption.type)
+      );
+    }
+
+    const subcommandName = message.content.split(' ')[1];
+    /** The command details object to use */
+    const commandDetails = this.details.subcommandDetails[subcommandName] ?? this.details;
+
     try {
-      const successResponse = await this.executeCommand(client, message);
+      const successResponse = await commandDetails.executeCommand(client, message, undefined, args);
```

Comment:
With `initialMessageFromBot` and `args` switched around, I guess the code will look cleaner in any case, since you can just leave out the `undefined` and do `const successResponse = await commandDetails.executeCommand(client, message, args);`.